src/main/scala/strawman/collection/immutable/List.scala

```diff
@@ -364,7 +364,17 @@ object List extends SeqFactory[List] {
   def empty[A]: List[A] = Nil
 
   private[collection] val partialNotApplied = new Function1[Any, Any] { def apply(x: Any): Any = this }
-
+  
+  def fillNew[A](n: Int)(elem: => A): List[A] = {
+    var filled = empty[A]
+    var i = 0
+    while (i < n) {
+      filled = ::(elem, filled)
```

Comment:
This fills it backwards, which breaks the (at least implied) contract of `fill` for state-dependent operations.